src/runtime/internal/lifecycle.ts

```diff
@@ -31,7 +31,7 @@ export function beforeUpdate(fn: () => any) {
  * 
  * https://svelte.dev/docs#run-time-svelte-onmount
  */
-export function onMount(fn: () => any) {
+export function onMount<T>(fn: (() =>  T extends Promise<void> ? T : T extends Promise<unknown> ? never : T)): void {
```

Comment:
I think we can simplify this to
```suggestion
export function onMount<T>(fn: (() => T extends Promise<() => any> ? never : T)): void {
```

Reply:
Thank you. updated. (Also I removed needless brackets.)